### PR TITLE
Dry run now returns sampler via debug dictionary entry

### LINKED ISSUE #187: Should dry runs returns dicts?
Follow up from #186 

In my opinion dry runs shouldn't have a return type that is similar to the expected return type. Ideally it should really just be `None` or `True`.

### PR DESCRIPTION
**Pull request by IAlibay**
Fixes #187 

The MultiStateSampler should contain everything we need for the openmm rbfe protocols, especially the HTF class that then also includes pointers to the individual component A and B systems / topologies.

### CODE REVIEW

File: openfe/tests/setup/test_openmm_equil_protocols.py [hunk 1]
```diff
@@ -148,15 +149,16 @@ def test_dry_run_default_vacuum(benzene_vacuum_system, toluene_vacuum_system,
...
 
     # create DAG from protocol and take first (and only) work unit from within
-    dag = protocol.create(
-        stateA=benzene_vacuum_system,
-        stateB=toluene_vacuum_system,
-        mapping={'ligand': benzene_to_toluene_mapping},
-    )
-    unit = list(dag.protocol_units)[0]
-
     with tmpdir.as_cwd():
```
Comment (IAlibay) [inline openfe/tests/setup/test_openmm_equil_protocols.py:152]: We need to be careful with our tmpdir locations, as far as I understand just calling `protocol.create` will create directories.

Comment (richardjgowers) [reply openfe/tests/setup/test_openmm_equil_protocols.py:152]: Yeah I'm getting dirs pop up during testing but it won't be in `create()` afaik... I'll track them down

**Review by richardjgowers — CHANGES_REQUESTED** [review]
this is a little overzealous with tmpdir usage, can I take this over @ialibay?

**Review by richardjgowers — APPROVED** [review]

**Comment by IAlibay** [comment]
ah nvm, just saw your recent commit

**Pull request merged**